qiskit/opflow/list_ops/list_op.py

```diff
@@ -101,6 +106,15 @@ def _state(
             "grad_combo_fn": grad_combo_fn if grad_combo_fn is not None else self.grad_combo_fn,
         }
 
+    @property
+    def settings(self) -> Dict:
+        """Return settings."""
+        if type(self) is ListOp and (
+            not self._using_default_combo_fn or self._grad_combo_fn is not None
```

Comment:
Would it make sense to always return the settings including the combo_fn and grad_combo_fn, since those really are the settings of the ListOp and check in the runtime encoder whether any of the arguments are callables? That would make the `settings` property more generic and not tailored specifically for JSON serialization 🙂

Reply:
Also, in this form, calling `settings` on e.g. `SummedOp` will always raise an error because it uses a different default combo function

Reply:
ok, 87cce54 made ListOp `settings` "generic" by including `combo_fn` and `grad_combo_fn`. However, this means its children need their own `settings` because the keywords in `settings` are meant to match the ones in `__init__`.